examples/js/ThreeLoader.js

```diff
@@ -0,0 +1,78 @@
+/* global itowns, Promise */
+var manager = new itowns.THREE.LoadingManager();
+
+// namespace ThreeLoader.
+var ThreeLoader = {};
+
+// Utility method to use Promise.
+function defer() {
+    var deferedPromise = {};
+    deferedPromise.promise = new Promise(function _(resolve, reject) {
+        deferedPromise.resolve = resolve;
+        deferedPromise.reject = reject;
+    });
+    return deferedPromise;
+}
+
+/**
+ * Load a script asynchronously
+ * @param {String} uri URI of the script to load.
+ * @returns {Promise} A promise resolved when the script is loaded.
+ */
+function loadScriptAsync(uri) {
+    var deferedPromise = defer();
+    var firstScriptTag = document.getElementsByTagName('script')[0];
+    var tag = document.createElement('script');
+    tag.async = true;
+    tag.onload = function r() {
+        deferedPromise.resolve();
+    };
+    tag.src = uri;
+    firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
+
+    return deferedPromise.promise;
+}
+
+/**
+ * Load script asynchronously from ThreeJs example loaders
+ *
+ * @param {String} format Format to be loader. Example: Collada will return ColladaLoader
+ * @returns {Promise} A promise with the ThreeJS Loader
+ */
+ThreeLoader.getThreeJsLoader = function getThreeJsLoader(format) {
+    var deferedPromise = defer();
+    // eslint-disable-next-line no-undef
+    THREE = itowns.THREE;
+    loadScriptAsync('https://cdn.rawgit.com/mrdoob/three.js/r' + itowns.THREE.REVISION + '/examples/js/loaders/' + format + 'Loader.js')
+    .then(function createLoader() {
+        deferedPromise.resolve(new itowns.THREE[format + 'Loader'](manager));
+    }).catch(function error(e) {
+        console.error('Error creating', format, 'loader : ', e);
+    });
+    return deferedPromise.promise;
+};
+
+/**
+ * Load a ressource asynchronously using an already created loader
+ * @param {Loader} loader Loader from ThreeJs example loaders
+ * @param {String} url URL of the ressource
+ * @returns {Promise} A promise resolved when the ressource is loaded.
+ */
+ThreeLoader.useThreeJsLoader = function useThreeJsLoader(loader, url) {
+    var deferedPromise = defer();
+    loader.load(url, deferedPromise.resolve, function _() {}, deferedPromise.reject);
+    return deferedPromise.promise;
+};
+
+/**
+ * Load a ressource asynchronously using ThreeJs example loader.
+ * @param {String} format Format to load. Format can be one of these : https://github.com/mrdoob/three.js/tree/dev/examples/js/loaders
```

Comment:
This needs to be more precise, like the user needs to take the name of the loader file minus `Loader.js` or something like that.

Reply:
yep